src/utils/wallet.ts

```diff
@@ -99,3 +99,5 @@ export const registerToken = async (
 
   return tokenAdded
 }
+
+export const canRegisterToken = () => typeof window !== 'undefined' && window?.ethereum?.isMetaMask
```

Comment:
does it need to be a function?

Reply:
let's keep it to make sure it's re-evaluated on client side